bindings/wasm/src/tangle/client.rs

```diff
@@ -255,16 +265,55 @@ impl Client {
   #[wasm_bindgen(js_name = checkPresentation)]
   pub fn check_presentation(&self, data: &str, options: WasmVerifierOptions) -> Result<Promise> {
     let client: Rc<IotaClient> = self.client.clone();
-    let data: Presentation = Presentation::from_json(&data).wasm_result()?;
-
+    let presentation: Presentation = Presentation::from_json(&data).wasm_result()?;
+    let presentation_validation_options =
+      PresentationValidationOptions::default().presentation_verifier_options(options.0);
     let promise: Promise = future_to_promise(async move {
-      CredentialValidator::new(&*client)
-        .validate_presentation(data, options.0)
-        .await
+      // resolve the holder's DID Document and also the DID Documents of the credential issuers.
+      let (holder_doc, issuer_docs): (ResolvedIotaDocument, Vec<ResolvedIotaDocument>) =
+        fetch_holder_and_issuers(client, &presentation).await.wasm_result()?;
+      let fail_fast: bool = true;
+      PresentationValidator::new()
+        .full_validation(
+          &presentation,
+          &presentation_validation_options,
+          &holder_doc,
+          issuer_docs.as_slice(),
+          fail_fast,
+        )
         .wasm_result()
-        .and_then(|output| JsValue::from_serde(&output).wasm_result())
+        .map(|_| JsValue::TRUE)
     });
 
     Ok(promise)
   }
 }
+
+// helper method for credential validation. Will be removed when we get the new Resolver.
+async fn resolve_did(
+  client: Rc<IotaClient>,
+  url: &str,
+) -> std::result::Result<ResolvedIotaDocument, identity::iota::Error> {
+  let did: IotaDID = url.parse()?;
+  let resolved_document: ResolvedIotaDocument = client.resolve(&did).await?;
+  Ok(resolved_document)
+}
+// helper method for presentation validation. Will be removed when we get the new Resolver.
```

Comment:
> We can discuss whether to merge the Resolver as-is so this PR can build on it rather than the other way around since there are quite a few instances of this.

Would be happy to make this PR build on the Resolver instead (and then also incorporate validation in the Wasm bindings as well). How far along is the work on the new resolver?

Reply:
Working on it now. Just needs Wasm bindings and a review pass to address things like its builder etc.

Reply:
#594 is merged and on dev now.

I imagine the high-level functions will be something like:

```
pub async fn verify_credential(credential: &Credential, trusted_issuers: Option<&[ResolvedIotaDocument]>) -> Result<()>;

pub async fn verify_presentation(presentation: &Presentation, trusted_issuers: Option<&[ResolvedIotaDocument]>, holder: Option<&ResolvedIotaDocument>) -> Result<()>;
```

There are problems taking slices/arrays as function parameters in `wasm-bindgen`, however.